static/breadCrumbs.tsx

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Rustw Project Developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+import * as React from 'react';
+
+interface IBreadCrumbProps {
+    getSource: (path: string, lineStart?: number) => void,
+    path: string
+}
+
+interface IBreadCrumbState {
+    
+}
+
+export default class BreadCrumbs extends React.Component<IBreadCrumbProps, IBreadCrumbState> {
```

Comment:
Does using TypeScript mean we can't use the function version of React components?

Reply:
No, I'll change it back.